### ISSUE TITLE
With GHS disabled table's styles are dropped when the figure has any style applied

### ISSUE THREAD

**Issue by dufipl** [ckeditor/ckeditor5#12987]
## 📝 Provide detailed reproduction steps (if any)

1. With GHS disabled, enter Source mode and paste the following code:
```
<figure class="table" style="width:75%;">
    <table style="height:75%;">
        <tbody>
            <tr>
                <td>
                    A cell
                </td>
            </tr>
        </tbody>
    </table>
</figure>
```
2. Apply this code and open source mode again.

### ✔️ Expected result

Both styles are applied:
<img width="903" alt="Zrzut ekranu 2022-12-7 o 13 01 32" src="https://user-images.githubusercontent.com/32898551/206174369-555ffc82-1d4e-436d-838f-006ea728def6.png">
e.g. like here: https://ckeditor.com/docs/ckeditor5/35.3.2/examples/builds-custom/full-featured-editor.html

### ❌ Actual result

The table style is dropped:
<img width="907" alt="Zrzut ekranu 2022-12-7 o 13 02 15" src="https://user-images.githubusercontent.com/32898551/206174505-a0cc1c6a-208b-4b2f-9675-664b472cfd30.png">

## 📃 Other details

It is a regression introduced probably by this PR: https://github.com/ckeditor/ckeditor5/pull/12948

* Browser: Any
* OS: Mac OS Monterey

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by niegowski** [comment]
It was handled this way on purpose (if there is a `figure` element then it uses styles from it, otherwise uses `table` styles), the tested code (in the reported issue) is never produced by CKE5 and IMHO it makes no sense. The approach there was pragmatic, why bother with messed up content when IFW or CKE5 does not generate it? Note that such attributes can be still handled by GHS.

**Comment by Reinmar** [comment]
I agree that it's fine for now. Let's wait  for a realistic case and revisit this decision if needed.
